stub/src/main/java/io/grpc/stub/ClientCallStreamObserver.java

```diff
@@ -24,4 +24,10 @@
  */
 @ExperimentalApi("https://github.com/grpc/grpc-java/issues/1788")
 public abstract class ClientCallStreamObserver<V> extends CallStreamObserver<V> {
+    /**
+     * Prevent any further processing for this {@code ClientCallStreamObserver}. No further messages will be received.
```

Comment:
No further messages may be sent either. If someone calls `clientCall.onNext()` after this it will throw.